### PR TITLE
Set openlineage provider as ready to be released

### PR DESCRIPTION
**Pull request by eladkal**
We are ready to prepare the provider wave before cutting 2.7.0 thus this is the time to add openlineage to release cycle

### CODE REVIEW

File: dev/provider_packages/prepare_provider_packages.py [hunk 1]
```diff
@@ -1676,10 +1676,12 @@ def verify_changelog_exists(package: str) -> str:
...
-    if "openlineage" in providers:
-        providers.remove("openlineage")
+    # if provider needs to be not considered in release add it here
+    # this is useful for cases where provider is WIP for a long period thus we don't want to release it yet.
+    providers_to_remove_from_release = []
+    for provider in providers_to_remove_from_release:
+        if provider in providers:
+            providers.remove(provider)
     for provider in providers:
         console.print(provider)
```
Comment (uranusjr) [inline dev/provider_packages/prepare_provider_packages.py]: Why do the in-place modification? We can just stick a `if provider in providers_to_remove_from_release: continue` in the latter for loop instead.

Comment (eladkal) [reply dev/provider_packages/prepare_provider_packages.py]: It's not the same list.
We have list of all providers and we have providers to exclude. I just converted the providers to exclude into a list.

Comment (uranusjr) [reply dev/provider_packages/prepare_provider_packages.py]: My question was why you need to modify the list at all. I was suggesting

```python
for provider in providers:
    if provider in providers_to_remove_from_release:
        continue
    console.print(provider)
```

**Review by mobuchowski — APPROVED** [review]
Let's do it 🙂

**Review by hussein-awala — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**